jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/CompoundPool.java

```diff
@@ -0,0 +1,147 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.util;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+/**
+ * This implementation does not support multiplexing.
+ *
+ * @param <P> the type of the pooled objects
+ */
+public class CompoundPool<P> implements Pool<P>
+{
+    private final ConcurrentPool<P> concurrentPool;
+    private final Pool<P> secondBestPool;
+
+    public CompoundPool(ConcurrentPool.StrategyType strategyType, int maxSize, boolean cache)
+    {
+        if (maxSize <= ConcurrentPool.MAX_RECOMMENDED_SIZE)
+        {
+            this.concurrentPool = new ConcurrentPool<>(strategyType, maxSize, cache);
+            this.secondBestPool = new NullPool<>();
+        }
+        else
+        {
+            this.concurrentPool = new ConcurrentPool<>(strategyType, ConcurrentPool.MAX_RECOMMENDED_SIZE, cache);
+            this.secondBestPool = new QueuedPool<>(maxSize - ConcurrentPool.MAX_RECOMMENDED_SIZE);
+        }
```

Comment:
That's a can of worms. I considered adding a `isMultiplexing` method to `Pool` but then I realized that multiplexing is based on a constructor-provided `ToIntFunction` which makes it impossible to be 100% certain both pools are going to use the same, or compatible multiplexing functions.

The best I can think of is to have such `isMultiplexing` method in `Pool` and make `ConcurrentPool` return `false` only when the constructor that doesn't take a `ToIntFunction` has been used.

Reply:
that works for me.

Reply:
If we introduce another pool that can handle queued multiplexed pools, then it will get more usage.   It is a util class and I see no problem exposing it.  All util classes can be used badly.   If we provide some static factory methods that easily build combinations that work well together, then that will encourage good usage.        I am OK with a protected constructor an only public static factory methods.  Then if somebody really wanted to do something strange, they could extend.